src/ringct/rctSigs.cpp

```diff
@@ -1579,13 +1579,31 @@ namespace rct {
       }
     }
 
-    bool verRctNonSemanticsSimpleCached(const rctSig & rv)
+    // I put docstring below in case this function gets refactored into another file
+    /**
+     * @brief This is cached version of verRctNonSemanticsSimple which accepts an expanded TX
+     *
+     * All non-serialized fields of tx.rct_signatures should match values present in tx. More
+     * specifically, the key images at tx.rct_signatures.CLSAGs[x].I should match tx.vin[x]. Also,
+     * tx.rct_signatures.bulletproofs_plus[x].V should match values tx.rct_signatures_outPk
+     * (multiplied by group inverse).
+     *
+     * @param tx transaction as seen externally which contains signature (with BC data) to verify
+     * @return true if test passes
+     * @return false if test does not pass
+     */
+    bool verRctNonSemanticsSimpleCached(const cryptonote::transaction & tx)
     {
+      const rctSig & rv = tx.rct_signatures;
+
       // Hello future Monero dev! If you got this assert, read the following carefully:
       //
-      // RCT cache assumes that this function will serialize and hash all rv's fields used for RingCT verification
-      // If you're about to add a new RCTType here, first you must check that binary_archive serialization writes all rv's fields to the binary blob
-      // If it's not the case, rewrite this function to serialize everything, even some "temporary" fields which are not serialized normally
+      // RCT cache assumes that the the transaction passed is the one in which the ring signature
```

Comment:
Some grammar and clarity issues in this comment. Would suggest:
```
// ::do_serialize(rct::rctSig) omits certain transaction-linked rct::rctSig
// fields which affect the semantics of the underlying signature. Therefore,
// the transaction hash must be included in the cache key so that caching
// a rct::rctSig, modifying its transaction-linked fields, and then performing
// a cache lookup on the modified signature, does not yield a false cache hit. 
```